src/note_encryption/orchard_domain.rs

```diff
@@ -0,0 +1,113 @@
+//! The OrchardDomain trait represents the difference between the `OrchardVanilla` and the `OrchardZSA`
+//! encryption and decryption procedures.
+
+use core::fmt;
+
+use zcash_note_encryption_zsa::{AEAD_TAG_SIZE, MEMO_SIZE};
+
+use crate::{action::Action, note::Rho, Note};
+
+use super::{compact_action::CompactAction, domain::Memo};
+
+/// Represents a fixed-size array of bytes for note components.
+#[derive(Clone, Copy, Debug)]
+pub struct NoteBytesData<const N: usize>(pub [u8; N]);
+
+impl<const N: usize> AsRef<[u8]> for NoteBytesData<N> {
+    fn as_ref(&self) -> &[u8] {
+        &self.0
+    }
+}
+
+impl<const N: usize> AsMut<[u8]> for NoteBytesData<N> {
+    fn as_mut(&mut self) -> &mut [u8] {
+        &mut self.0
+    }
+}
+
+// FIXME: consider implementing and using TryFrom instead (see also review comments in
+// zcash_note_encryption PR)
```

Comment:
remove comment. We have it under a different task

Reply:
Removed.